devicetypes/stelpro/stelpro-maestro-thermostat.src/stelpro-maestro-thermostat.groovy

```diff
@@ -333,18 +336,26 @@ def handleTemperature(descMap) {
 		def lastTemp = device.currentValue("temperature")
 		def lastAlarm = device.currentValue("temperatureAlarm")
 		if (lastAlarm != "cleared") {
+			// Should we? lastTemp = convertTempIfNeeded
+			def cleared = false
 			if (lastTemp && device.currentState("temperature").unit == map.unit) {
 				// Check to see if we are coming out of our alarm state and only clear then
 				// NOTE: A thermostat might send us an alarm *before* it has completed sending us previous measurements,
 				// so it might appear that the alarm is no longer valid. We need to check the trajectory of the temperature
 				// to verify this.
 				if ((lastAlarm == "freeze" &&
-						map.value > (map.unit == "C" ? 0 : 32) &&
+						map.value > FREEZE_ALARM_TEMP &&
 						lastTemp < map.value) ||
 					(lastAlarm == "heat" &&
-						map.value < (map.unit == "C" ? 50 : 122) &&
```

Comment:
isn't it possible that the device-reported unit is different than the user's units (`getTemperatureScale()`)?

Reply:
They should match. map.unit is set to getTemperatureScale, and Zigbee devices report in C and we convert map.value when map.unit is set.